pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -158,8 +158,12 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
         }
 
         // Initialize inverted index creator
+        // Do not created inverted index if sorted, clean up invertedIndexColumns
         if (invertedIndexColumns.contains(columnName)) {
```

Comment:
Change this line to `if (invertedIndexColumns.contains(columnName) && !indexCreationInfo.isSorted())` should be good. We don't need to clean up the invertedIndexColumns (it won't be populated outside even if you changed it).